### PR TITLE
Ensured that job_class values ScheduledJob instances are transferred to task during v2 migration.

### PR DESCRIPTION
**Pull request by HanlinMiao**
# Closes: #4676 
# What's Changed

# TODO

- [x] Explanation of Change(s)
- [x] Added change log fragment(s) (for more information see [the documentation](https://docs.nautobot.com/projects/core/en/stable/development/#creating-changelog-fragments))
- [ ] Attached Screenshots, Payload Example
- [ ] Unit, Integration Tests
- [ ] Documentation Updates (when adding/changing features)
- [ ] Example Plugin Updates (when adding/changing features)
- [ ] Outline Remaining Work, Constraints from Design

### CODE REVIEW

File: nautobot/extras/migrations/0092_uniqueness_data_migration.py [hunk 1]
```diff
@@ -3,11 +3,13 @@
...
 
 
-def ensure_unique_scheduledjob_names(apps, schema_editor):
+def ensure_unique_scheduledjob_names_and_save_job_class_parameter_to_task(apps, schema_editor):
     ScheduledJob = apps.get_model("extras", "ScheduledJob")
 
     cache = set()
     for instance in ScheduledJob.objects.all():
+        instance.task = instance.job_class
+        instance.save()
```
Comment (glennmatthews) [inline nautobot/extras/migrations/0092_uniqueness_data_migration.py:12]: Was there a reason this couldn't/shouldn't be done in 0091 (where we're updating `job_class` to 2.0 standards) instead? This is probably fine, just wondering.

File: changes/4676.changed
Comment (glennmatthews) [inline changes/4676.changed:1]: Seems like it should be a `.fixed` to me?

File: nautobot/extras/migrations/0092_uniqueness_data_migration.py (hunk 1 above)
Comment (HanlinMiao) [reply nautobot/extras/migrations/0092_uniqueness_data_migration.py]: Ur right, it makes more sense to include it in 0091.

File: changes/4676.fixed [hunk 2]
```diff
@@ -0,0 +1 @@
+Ensured that `job_class` values `ScheduledJob` instances are transferred to `task` during v2 migration.
```
Comment (glennmatthews) [inline changes/4676.fixed:1]:
```suggestion
Ensured that `ScheduledJob.job_class` values are correctly transferred to `ScheduledJob.task` during v2 migration.
```

File: nautobot/extras/migrations/0092_uniqueness_data_migration.py [hunk 3]
```diff
@@ -3,7 +3,7 @@
 from nautobot.core.utils.migrations import check_for_duplicates_with_natural_key_fields_in_migration
 
 
-def ensure_unique_scheduledjob_names(apps, schema_editor):
+def ensure_unique_scheduledjob_names_and_save_job_class_parameter_to_task(apps, schema_editor):
```
Comment (glennmatthews) [inline nautobot/extras/migrations/0092_uniqueness_data_migration.py:6]: Revert?

File: nautobot/extras/migrations/0092_uniqueness_data_migration.py [hunk 4]
```diff
@@ -40,6 +40,8 @@ class Migration(migrations.Migration):
     ]
 
     operations = [
-        migrations.RunPython(ensure_unique_scheduledjob_names, migrations.RunPython.noop),
+        migrations.RunPython(
+            ensure_unique_scheduledjob_names_and_save_job_class_parameter_to_task, migrations.RunPython.noop
+        ),
```
Comment (glennmatthews) [inline nautobot/extras/migrations/0092_uniqueness_data_migration.py:45]: Revert?

File: nautobot/extras/migrations/0092_uniqueness_data_migration.py (hunk 3 above)
Comment (HanlinMiao) [reply nautobot/extras/migrations/0092_uniqueness_data_migration.py]: Idk how I missed that sorry

**Review by glennmatthews — APPROVED** [review]

**Pull request merged**